src/main/java/com/artipie/scripting/ScriptContext.java

```diff
@@ -0,0 +1,191 @@
+        final long duration = new Property(ArtipieProperties.SCRIPTS_TIMEOUT)
+            .asLongOrDefault(120_000L);
+        return CacheBuilder.newBuilder()
+            .expireAfterWrite(duration, TimeUnit.MILLISECONDS)
+            .softValues()
+            .build(
+                new CacheLoader<>() {
+                    @Override
+                    public Script.PrecompiledScript load(final FilesContent ctx) {
+                        return ctx.precompiledScript();
+                    }
+                }
+            );
+    }
+
+    /**
+     * Extra class for caching precompiled scripts.
+     * @since 0.1
+     */
+    static final class FilesContent {
+        /**
+         * Key.
+         */
+        private final Key key;
+
+        /**
+         * Storage.
+         */
+        private final BlockingStorage storage;
+
+        /**
+         * Ctor.
+         * @param key Key
+         * @param storage Storage
+         */
+        FilesContent(final Key key, final BlockingStorage storage) {
+            this.key = key;
+            this.storage = storage;
+        }
+
+        @Override
+        public int hashCode() {
+            return this.key.hashCode();
+        }
+
+        @Override
+        public boolean equals(final Object obj) {
+            final boolean res;
+            if (obj == this) {
+                res = true;
+            } else if (obj instanceof ScriptContext.FilesContent) {
+                final ScriptContext.FilesContent data = (ScriptContext.FilesContent) obj;
+                res = Objects.equals(this.key, data.key);
+            } else {
+                res = false;
+            }
+            return res;
+        }
+
+        /**
+         * Returns precompiled script for stored key & storage.
+         * @return Returns precompiled script object instance.
+         */
+        public Script.PrecompiledScript precompiledScript() {
+            final String ext = FilesContent.extension(this.key.string());
+            final Script.ScriptType type = Arrays.stream(Script.ScriptType.values())
+                .filter(val -> val.ext().equals(ext)).findFirst().orElse(Script.ScriptType.NONE);
+            Script.PrecompiledScript result = null;
+            if (!type.equals(Script.ScriptType.NONE)) {
+                final String script = new String(this.storage.value(this.key));
+                result = new Script.PrecompiledScript(type, script);
+            }
+            return result;
+        }
+
+        /**
+         * Obtain extension of filename.
+         * @param filename Name of file.
+         * @return Extension.
+         */
+        private static String extension(final String filename) {
+            final int pos = filename.lastIndexOf('.');
+            final String res;
+            if (pos >= 0) {
+                res = filename.substring(pos + 1);
+            } else {
+                res = "";
+            }
+            return res;
```

Comment:
Cannot we use some standard method to obtain the extension? `FilenameUtils.getExtension` from commons.io should be available